src/ResourceManager/RecoveryServices/Commands.RecoveryServices.Backup.Providers/Providers/AzureFilesPsBackupProvider.cs

```diff
@@ -139,29 +382,6 @@ public RestAzureNS.AzureOperationResponse EnableProtection()
                 resourceGroupName: vaultResourceGroupName);
         }
 
-        private void ValidateAzureFilesWorkloadType(CmdletModel.WorkloadType type)
-        {
-            if (type != CmdletModel.WorkloadType.AzureFiles)
-            {
-                throw new ArgumentException(string.Format(Resources.UnExpectedWorkLoadTypeException,
-                                            CmdletModel.WorkloadType.AzureFiles.ToString(),
-                                            type.ToString()));
-            }
-        }
-
-        private void ValidateAzureFilesWorkloadType(CmdletModel.WorkloadType itemWorkloadType,
-            CmdletModel.WorkloadType policyWorkloadType)
-        {
-            ValidateAzureFilesWorkloadType(itemWorkloadType);
-            ValidateAzureFilesWorkloadType(policyWorkloadType);
-            if (itemWorkloadType != policyWorkloadType)
-            {
-                throw new ArgumentException(string.Format(Resources.UnExpectedWorkLoadTypeException,
-                                            CmdletModel.WorkloadType.AzureFiles.ToString(),
-                                            itemWorkloadType.ToString()));
-            }
-        }
-
```

Comment:
In this method, you are throwing an incorrect exception message with **VirtualMachineIdIsEmptyOrNull**. Please fix this:

         private void ValidateAzureFilesModifyProtectionRequest(ItemBase itemBase,
             PolicyBase policy)
         {
             if (itemBase == null || itemBase.GetType() != typeof(AzureFileShareItem))
             {
                 throw new ArgumentException(string.Format(Resources.InvalidProtectionPolicyException,
                                             typeof(AzureFileShareItem).ToString()));
             }
 
             if (string.IsNullOrEmpty(((AzureFileShareItem)itemBase).ParentContainerFabricId))
             {
                 throw new ArgumentException(Resources.VirtualMachineIdIsEmptyOrNull);
             }
         }